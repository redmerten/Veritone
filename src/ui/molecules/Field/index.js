// @flow

import { rem } from 'polished';
import React, { type Node } from 'react';
import styled, { css } from 'styled-components';

import Label from '../../atoms/Label';

type Props = {
  children?: Node,
  className: string,
  dataTestId?: string,
  info?: Node,
  labelFor?: string,
  label?: string,
};

const Container = ({ children, className, dataTestId = 'molecule-field', info, label, labelFor }: Props) => (
  <div className={className} data-testid={dataTestId}>
    {info && info}
    {label && <Label htmlFor={labelFor}>{label}</Label>}
    {children}
  </div>
);

const Field = styled(Container)`
  align-items: center;
  display: flex;
  flex-direction: row;
  margin-top: ${({ theme }) => rem(theme.spacing.md)};
  width: 100%;

  ${({ label, theme }) =>
    !label &&
    css`
      padding-left: ${rem(140 + theme.spacing.md)};
    `};

  /* stylelint-disable-next-line */
  ${Label} {
    flex-shrink: 0;
    margin-right: ${({ theme }) => rem(theme.spacing.md)};
    min-width: ${({ info }) => (info ? rem(125) : rem(140))};
    max-width: ${({ info }) => (info ? rem(125) : rem(140))};
  }

  ${({ stacked }) =>
    stacked &&
    css`
      flex-direction: column;
      justify-content: flex-start;
      width: ${rem(400)};

      /* stylelint-disable-next-line */
      ${Label} {
        align-self: flex-start;
        margin-bottom: ${({ theme }) => rem(theme.spacing.xs)};
        width: auto;
      }
    `};
`;

Field.displayName = 'Field';
export default Field;
