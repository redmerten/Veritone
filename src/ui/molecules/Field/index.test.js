// @flow

import React from 'react';

import Field from '.';

import { render } from 'utils/testing';

describe('molecules/Field', () => {
  it('matches its base snapshot', () => {
    const { baseElement } = render(<Field className="class-name" />);
    expect(baseElement.firstChild).toMatchSnapshot();
  });

  it('renders Field molecule', () => {
    const { getByTestId } = render(<Field className="class-name" />);
    expect(getByTestId('molecule-field')).toBeInTheDocument();
  });

  it('should render with label', () => {
    const { getByText } = render(<Field className="class-name" label="Label here" />);
    expect(getByText('Label here')).toBeInTheDocument();
  });

  it('renders its children', () => {
    const { getByText } = render(
      <Field className="class-name">
        <p>Children here</p>
      </Field>
    );

    expect(getByText('Children here')).toBeInTheDocument();
  });
});
