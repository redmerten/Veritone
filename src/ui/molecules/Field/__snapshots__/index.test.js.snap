// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/Field matches its base snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: 1rem;
  padding-left: 9.75rem;
}

.c0 .c1 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1rem;
  min-width: 8.75rem;
  max-width: 8.75rem;
}

.c0 .c2 {
  width: auto;
}

<div>
  <div
    class="c0 class-name"
    data-testid="molecule-field"
  />
</div>
`;
